crates/metal-codegen-llvm/src/safeties.rs

```diff
@@ -33,3 +36,36 @@ impl Drop for MemoryBuffer {
         unsafe { LLVMDisposeMemoryBuffer(self.llvm_buf) };
     }
 }
+
+pub struct LLVMErrorMessage {
+    ptr: *mut c_char,
+}
+
+impl Default for LLVMErrorMessage {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl LLVMErrorMessage {
+    pub fn new() -> Self {
+        Self {
+            ptr: std::ptr::null_mut(),
```

Comment:
i don't feel too good about this, i feel like some kind of buffer needs to be allocated, but i couldn't understand what the llvmverifymodule function does so this is gonna do for now

Reply:
> i don't feel too good about this, i feel like some kind of buffer needs to be allocated, but i couldn't understand what the llvmverifymodule function does so this is gonna do for now

This is what is shown in all examples I could find. It works and doesn't leak memory so, there's no problems.